### ISSUE TITLE
Keep alive stream prematurely EOFs

### ISSUE THREAD

**Issue by tobyWorland** [fukamachi/dexador#139]
Hello :wave: 

I am attempting to read from this url: https://raw.githubusercontent.com/vega/vega/main/docs/data/gapminder.json

There are 68753 characters in the file, all on a single line.

The following return the correct amount:
```
(let ((s (dex:get "https://raw.githubusercontent.com/vega/vega/main/docs/data/gapminder.json" :want-stream t :keep-alive nil)))
	   (prog1 (length (read-line s))
	     (close s)))
```
```
(length (dex:get "https://raw.githubusercontent.com/vega/vega/main/docs/data/gapminder.json"))
```
This one returns 68750:
```
(let ((s (dex:get "https://raw.githubusercontent.com/vega/vega/main/docs/data/gapminder.json" :want-stream t)))
	   (length (read-line s)))
```
The bug seems to have been introduced in commit 05e5c5ae974e2af19c20a983e3e84f986bd855be

**Comment by ajberkley** [comment]
Thank you.  Fix in progress.

**Comment by ajberkley** [comment]
Closed by #140  Will submit a test for this in a few more moments.

**Comment by tobyWorland** [comment]
Thanks!
